src/main/scala/org/constellation/primitives/ChannelMessage.scala

```diff
@@ -30,6 +42,34 @@ object ChannelMessage {
       SignedData(data, hashSignBatchZeroTyped(data, dao.keyPair))
     )
   }
+
+  def createGenesis(channelOpenRequest: ChannelOpenRequest)(implicit dao: DAO): Option[ChannelMessage] = {
+    if (
+      dao.messageService.get(channelOpenRequest.channelId).isEmpty &&
+      !dao.threadSafeMessageMemPool.activeChannels.contains(channelOpenRequest.channelId)
+    ) Some {
+
+      val genesisMessageStr = ChannelOpen(channelOpenRequest.jsonSchema, channelOpenRequest.acceptInvalid).json
+      val msg = create(genesisMessageStr, Genesis.CoinBaseHash, channelOpenRequest.channelId)
+      val semaphore = new Semaphore(1)
+      dao.threadSafeMessageMemPool.activeChannels(channelOpenRequest.channelId) = semaphore
+      semaphore.acquire()
+      dao.threadSafeMessageMemPool.put(Seq(msg), overrideLimit = true)
+      msg
```

Comment:
to me it's a lot more readable when you just do Some(msg) at the end then nest the whole block like this.

Reply:
Will fix